pkg/clusterresource/aws.go

```diff
@@ -107,9 +107,19 @@ func (p *AWSCloudBuilder) addInstallConfigPlatform(o *Builder, ic *installertype
 			Type: volumeType,
 		},
 	}
-	ic.ControlPlane.Platform.AWS = mpp
-	ic.Compute[0].Platform.AWS = mpp
 
+	// Use supplied machinepool configuration if possible
+	if (o.ControlPlaneMachinePool != installertypes.MachinePoolPlatform{}) {
+		ic.ControlPlane.Platform.AWS = o.ControlPlaneMachinePool.AWS
+	} else {
+		ic.ControlPlane.Platform.AWS = mpp
+	}
+
+	if (o.WorkerMachinePool != installertypes.MachinePoolPlatform{}) {
+		ic.Compute[0].Platform.AWS = o.WorkerMachinePool.AWS
+	} else {
+		ic.Compute[0].Platform.AWS = mpp
+	}
```

Comment:
This covers the installconfig but I think you're then going to conflict with the Hive MachinePool that takes over day 2 operation, see addMachinePoolPlatform above. You'll want to update this to use the same settings. Per comments above the API should use the Hive MachinePoolPlatform anyhow so that one will be easy, but this part for the install config will be where you need to explicitly translate what we support.